### PR TITLE
Increasing unit test coverage for VariantContextConverter

### PR DESCRIPTION
**Pull request by heuermh**

### CODE REVIEW

**Comment by heuermh** [comment]
If I `ignore` the hanging unit test then I see VCF header-related exceptions
```
- don't lose any variants when piping as VCF !!! IGNORED !!!
2016-11-16 17:19:15 ERROR Utils:95 - Aborting task
java.lang.IllegalStateException: Key IndelQD found in VariantContext field FILTER at 1:14397 but this key isn't defined in the VCFHeader.  We require all VCFs to have complete VCF headers by default.
	at htsjdk.variant.vcf.VCFEncoder.fieldIsMissingFromHeaderError(VCFEncoder.java:173)
	at htsjdk.variant.vcf.VCFEncoder.getFilterString(VCFEncoder.java:154)
	at htsjdk.variant.vcf.VCFEncoder.encode(VCFEncoder.java:106)
	at htsjdk.variant.variantcontext.writer.VCFWriter.add(VCFWriter.java:222)
	at org.seqdoop.hadoop_bam.VCFRecordWriter.writeRecord(VCFRecordWriter.java:140)
	at org.seqdoop.hadoop_bam.KeyIgnoringVCFRecordWriter.write(KeyIgnoringVCFRecordWriter.java:60)
	at org.seqdoop.hadoop_bam.KeyIgnoringVCFRecordWriter.write(KeyIgnoringVCFRecordWriter.java:38)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12$$anonfun$apply$4.apply$mcV$sp(PairRDDFunctions.scala:1113)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12$$anonfun$apply$4.apply(PairRDDFunctions.scala:1111)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12$$anonfun$apply$4.apply(PairRDDFunctions.scala:1111)
	at org.apache.spark.util.Utils$.tryWithSafeFinallyAndFailureCallbacks(Utils.scala:1277)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12.apply(PairRDDFunctions.scala:1119)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12.apply(PairRDDFunctions.scala:1091)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:66)
	at org.apache.spark.scheduler.Task.run(Task.scala:89)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:227)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by fnothaft** [comment]
We'll need #1260 + a bit more to fix the header lines issue...

File: adam-core/src/test/scala/org/bdgenomics/adam/rdd/ADAMContextSuite.scala [hunk 1]
```diff
@@ -152,6 +152,22 @@ class ADAMContextSuite extends ADAMFunSuite {
     assert(vcs.size === 6)
 
     val vc = vcs.head
+
+    /*
```
Comment (fnothaft) [inline adam-core/src/test/scala/org/bdgenomics/adam/rdd/ADAMContextSuite.scala]: If all's the same to you, I'd nix this comment.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala [hunk 2]
```diff
@@ -607,6 +610,21 @@ private[adam] class VariantContextConverter(dict: Option[SequenceDictionary] = N
...
+    val filtersPassed = Option(variant.getFiltersPassed).getOrElse(false)
+
+    (filtersApplied, filtersPassed) match {
+      case (false, false) => vcb.unfiltered
+      case (false, true)  => vcb.passFilters // log warning?
+      case (true, false)  => vcb.filters(new java.util.HashSet(variant.getFiltersFailed()))
+      case (true, true)   => vcb.passFilters
+    }
+
+    val somatic: java.lang.Boolean = Option(variant.getSomatic).getOrElse(false)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala]: I'd lose the `: java.lang.Boolean`. Is there a reason you need it?

Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala]: Yeah it wouldn't compile without it. Odd that the lines above were ok.

**Comment by heuermh** [comment]
Yes, I believe that is what is happening with the hang. Teeing to another file results in an empty file.

File: adam-core/src/test/scala/org/bdgenomics/adam/converters/VariantContextConverterSuite.scala [hunk 3]
```diff
@@ -86,6 +86,27 @@ class VariantContextConverterSuite extends ADAMFunSuite {
 
     assert(variant.getReferenceAllele === "A")
     assert(variant.getStart === 0L)
+    assert(variant.getSomatic === false)
+  }
+
+  test("Convert somatic htsjdk site-only SNV to ADAM") {
+    val converter = new VariantContextConverter
+
+    // not sure why this doesn't work
```
Comment (fnothaft) [inline adam-core/src/test/scala/org/bdgenomics/adam/converters/VariantContextConverterSuite.scala]: This one too.

**Pull request merged**